File: testing/vector_2d.cpp

// 2D vector in polar coordinates.
// Angle must be in radians.

#include "vector_2d.hpp"


Cartesian_Vector_2D Polar_Vector_2D::make_cartesian()
{
    double x = radius * cos(angle);
    double y = radius * sin(angle);
    Cartesian_Vector_2D result {x, y};
    return result;
}


Quantized_Cartesian_Vector_2D Cartesian_Vector_2D::make_quantized()
{
    int x_quant = x; // implicit type narrowing :)
    int y_quant = y;
    Quantized_Cartesian_Vector_2D result {x_quant, y_quant};
    return result;
}

File: misc/old/vector_2d.py

"""
2D polar vector class.
"""

from math import cos, sin


class Polar_Vector_2D:
    """
    A 2D vector in polar coordinates.
    """
    def __init__(self, r, theta):
        self.r = r
        self.theta = theta
    
    def to_cartesian_x(self,):
        x = self.r * cos(self.theta)
        return x
    
    def to_cartesian_y(self,):
        y = self.r * sin(self.theta)
        return y